op-program/README.md

```diff
@@ -32,3 +32,29 @@ From within the `op-program` directory, options can be reviewed with:
 ```shell
 ./bin/op-program --help
 ```
+
+## Generating the Absolute Prestate
+
+The absolute pre-state of the op-program can be generated by executing the makefile
+`reproducible-prestate` target. Effectively, this builds a docker image specified
+by [Dockerfile.repro](./Dockerfile.repro) which uses a docker environment with the
+following pinned dependencies:
+- golang version `1.21.3`
+- alpine `3.18`
+
+Internally, [Dockerfile.repro](./Dockerfile.repro) builds the `op-program` using a
+mips32 big endian architecture and the `cannon` binary. It then passes in the
+`op-program` elf binary through cannon's `load-elf` subcommand to generate the
+`prestate.json` file. Finally, the `prestate.json` file is "proven" using cannon's
+`--proof-at=0`, outputting a `prestate-proof.json`. This `prestate-proof.json` file
+contains the absolute pre-state hash that we are looking for under the `pre` key.
```

Comment:
I think we can skip the dependencies and description of what the Dockerfile does - the Dockerfile itself documents that accurately.  The key instructions are just to run `make reproducible-prostate` and that the resulting prestate is in `op-program/bin/prestate-proof.json` and the `pre` field is the hash used in the contract deployment.